### PR TITLE
Add Google Analytic Events

### PR DESCRIPTION
**Pull request by mmcfarland**
## Overview

Adds Event tracking for high profile activity within the app, for the purposes of describing app user behavior to WPF.

* Aoi Creation: Captures the type of aoi creation selected: boundary type, freedraw, squarekm, drb-rwd, nhd-drb, shapefile or geojson
* Analyze: An analyze request for an AoI, one for each type: animals, land, soil, etc.
* Modeling: Captures creating a project, the model type, adding a scenario and adding a modification.  Scenario and Modification events capture the model package as well 

Connects #2450 

### Demo
Events showing up:
image

Event requests being sent (Network -> JS)
image

### Notes

There are a large number of potential events to capture, however I chose the top actions that I anticipated WPF may wish to know.  This could be expanded to determine which features are commonly used, but since we don't have explicit funding for that type of analysis, I'm keeping it as simple as possible for now.

## Testing Instructions

 * Application behavior should not have changed at all
 * GA Events take up to 24 hours to register in the dashboard, but Real-Time events can be seen almost immediately.  I can help set up access for you to test.

### CODE REVIEW

File: src/mmw/js/src/core/utils.js [hunk 1]
```diff
@@ -513,6 +513,13 @@ var utils = {
     // Array.filter(distinct) to get distinct values
     distinct: function(value, index, self) {
         return self.indexOf(value) === index;
+    },
+
+    isInDrb: function(geom) {
```
Comment (rajadain) [inline src/mmw/js/src/core/utils.js]: Surprising we didn't have this utility already

**Comment by rajadain** [comment]
Interestingly enough, I had to disable my Google Analytics Opt Out Chrome extension: https://chrome.google.com/webstore/detail/google-analytics-opt-out/fllaojicojecljbmefodhfapmkghcbnh to test this. There's probably a minuscule number of users out there with something like this in place though.

**Comment by kellyi** [comment]
Heads up that it looks like Jenkins gets hung up on this test:

```
not ok 105 Firefox 56.0 - Modeling Models ScenarioModel #updateModificationHash is called before model#attemptSave when the modifications change
    ---
        message: >
            ga is not defined
        stack: >
            addModification@http://localhost:7357/3130/js/test.bundle.js:21322:9
            [139]</</</</</<@http://localhost:7357/3130/js/test.bundle.js:24205:21
            Runnable.prototype.run@http://localhost:7357/testem/mocha.js:4212:24
            Runner.prototype.runTest@http://localhost:7357/testem/mocha.js:4584:5
            next/<@http://localhost:7357/testem/mocha.js:4630:7
            next@http://localhost:7357/testem/mocha.js:4510:14
            next/<@http://localhost:7357/testem/mocha.js:4519:7
            next@http://localhost:7357/testem/mocha.js:4463:23
            Runner.prototype.hook/<@http://localhost:7357/testem/mocha.js:4487:5
            timeslice@http://localhost:7357/testem/mocha.js:5480:5
            
    ...
```

**Comment by mmcfarland** [comment]
I specifically bundled and checked the tests thinking that `ga` would not be defined for them.  They passed locally but are failing in CI so I'm going to add a commit to define `ga` as `noop` in the tests.

**Comment by mmcfarland** [comment]
Going around again, check out 0a3198a for fixup that cleans up lint.  The linter didn't like the unaccounted for global var.

```
"js/src/draw/views.js: line 823, col 75, Missing semicolon.", 
"js/src/draw/views.js: line 423, col 9, 'ga' is not defined.", 
"js/src/draw/views.js: line 438, col 9, 'ga' is not defined.", 
"js/src/draw/views.js: line 671, col 17, 'ga' is not defined.", 
...
```

**Comment by rajadain** [comment]
Friendly reminder to squash before merging.

**Pull request merged**